examples/ACE/utils/NNIAP.jl

```diff
@@ -72,12 +130,45 @@ function learn!(nniap, ds, opt::Flux.Optimise.AbstractOptimiser, epochs, loss, w
         grads = ∇loss(nniap.nn, nniap.iap, ds, w_e, w_f)
         Flux.update!(optim, nniap.nn, grads[1])
         # Logging
-        curr_loss = loss(nniap.nn, nniap.iap, ds, w_e, w_f)
+        loss_time = @elapsed curr_loss = loss(nniap.nn, nniap.iap, ds, w_e, w_f)
+        println("loss_time", loss_time)
         push!(losses, curr_loss)
         println(curr_loss)
     end
 end
 
+function learn!(nace, ds_train, opt, n_epochs, n_batches, loss, w_e, w_f, _device)
+    nn = nace.nn |> _device
+    iap = nace.iap |> _device
+    ds = ds_train |> _device
+    optim = Flux.setup(opt, nn) |> _device
+    ∇loss(nn, iap, batch, true_energy, local_descriptors, w_e, w_f) = gradient((nn) -> loss(nn, iap, batch, true_energy, local_descriptors, w_e, w_f), nn)
```

Comment:
Please, remove `batch` from the interface. 

It will probably be necessary to add a vector with the number of atoms per configuration of the `local_descriptors` batch array.